packages/mermaid/src/diagrams/common/commonTypes.ts

```diff
@@ -0,0 +1,58 @@
+export interface RectData {
+  x?: number;
+  y?: number;
+  fill?: string;
+  width?: number;
+  height?: number;
+  stroke?: string;
+  class?: string;
+  color?: string | number;
+  rx?: number;
+  ry?: number;
+  attrs?: Record<string, string | number>;
+  anchor?: string;
```

Comment:
You sure every parameter is optional here?

Reply:
It was weird while working on this type, it was throw errors every so I sat everything to optional :)
After rethinking, Only some should:

- rx
- ry
- class
- anchor

https://github.com/mermaid-js/mermaid/blob/541ee1eade6ff4835ea27b2aaccb0e07888212db/packages/mermaid/src/diagrams/common/svgDrawCommon.ts#L47-L55

https://github.com/mermaid-js/mermaid/blob/541ee1eade6ff4835ea27b2aaccb0e07888212db/packages/mermaid/src/diagrams/common/svgDrawCommon.ts#L100-L111

Reply:
Yup, can merge if you make the changes.

Reply:
Fixed 99c175849, but I fixed more stuff so I'd recommend re-reviewing it.